server/leafnode.go

```diff
@@ -801,6 +811,17 @@ func (c *client) initLeafNodeSmap() {
 	}
 	acc.mu.RUnlock()
 
+	// Now check for gateway interest. Leafnodes will put this into
+	// the proper mode to propagate, but they are not held in the account.
+	gwsa := [16]*client{}
+	gws := gwsa[:0]
+	s.getOutboundGatewayConnections(&gws)
+	for _, cgw := range gws {
+		if ei, _ := cgw.gw.outsim.Load(accName); ei != nil {
```

Comment:
I believe that we don't need indeed to cgw.mu.Lock()/Unlock() here since cgw.gw is set when the object is initialized and I don't think I ever set it to nil. 
Same for accessing `ei.(*outsie).sl.All()` without checking if sl is nil or not. `sl` should be initialized when an entry is added in the map, at least in the "gateway-only" code path. Since we have now trigger some gateway code from leafnode conditions, I am not entirely sure that this is always the case.

Reply:
Will take a look. I can lock it down more.

Reply:
Will fix.

Reply:
Take a look now..